crates/viewer/re_viewer_context/src/view/view_context_system.rs

```diff
@@ -1,17 +1,44 @@
+use std::any::Any;
+
 use ahash::HashMap;
 
 use re_types::ViewClassIdentifier;
 
-use crate::{IdentifiedViewSystem, ViewQuery, ViewSystemExecutionError, ViewSystemIdentifier};
+use crate::{
+    IdentifiedViewSystem, ViewContext, ViewQuery, ViewSystemExecutionError, ViewSystemIdentifier,
+    ViewerContext,
+};
 
-use super::view_context::ViewContext;
+pub type ViewContextSystemStaticExecResult = Box<dyn Any + Send + Sync>;
 
 /// View context that can be used by view parts and ui methods to retrieve information about the scene as a whole.
 ///
 /// Is always populated before view part systems.
 pub trait ViewContextSystem: Send + Sync {
+    /// Executes once per active _type_ of `ViewContextSystem`, independent of the view's state, query, blueprint properties etc.
+    ///
+    /// This is run each frame once per type of view context system if the context system is used by any view.
+    /// Afterwards, the such mutated context system instance gets a call to [`ViewContextSystem::execute`] for each view instance.
+    ///
+    /// Use this to perform any operations that are shared across all views that use this system,
+    /// independent of their state, query, blueprint properties etc.
+    fn execute_static(_ctx: &ViewerContext<'_>) -> ViewContextSystemStaticExecResult
+    where
+        Self: Sized,
+    {
+        Box::new(())
+    }
+
```

Comment:
I don't understand the "such mutated context system instance" part of the docstring. This method doesn't have a mutating receiver? How would the instance be mutated?